tapdb/assets_store_test.go

```diff
@@ -677,16 +683,21 @@ func TestSelectCommitment(t *testing.T) {
 			// First, we'll create a new assets store and then
 			// insert the set of assets described by the asset
 			// descriptions.
-			_, assetsStore, _ := newAssetStore(t)
+			_, store, _ := newAssetStore(t)
 
-			assetGen.genAssets(t, assetsStore, tc.assets)
+			assetGen.genAssets(t, store, tc.assets)
 
 			// With the assets inserted, we'll now attempt to query
 			// for the set of matching assets based on the
 			// constraints.
-			selectedAssets, err := assetsStore.ListEligibleCoins(
-				ctx, tc.constraints,
-			)
+			var selectedAssets []*tapfreighter.AnchoredCommitment
+			err := store.WithCoinLock(func() error {
```

Comment:
Re the prior comments, we then wouldn't have to leak the details of the coin selection lock into the database level. Instead, we can have a higher level unit test that just gives a mocked implementation of `CoinLister`.